### PR TITLE
Add tests to check if the ZIO Runtime is propagated to req execution

### PR DESCRIPTION
**Pull request by kyri-petrou**
Followup of #2807, adding a couple of tests to ensure that we're properly propagating the bootstrapped runtime to the request execution

### CODE REVIEW

**Review by 987Nabil — APPROVED** [review]

**Comment by kyri-petrou** [comment]
@987Nabil it seems there's a flaky test that keeps failing. Is this a new test? I don't remember having it fail so many times in a row

**Comment by 987Nabil** [comment]
@kyri-petrou I had some issues on my PR too. I'll take a look soonish, but I am not a the PC rn

File: zio-http/jvm/src/test/scala/zio/http/FormSpec.scala [hunk 1]
```diff
@@ -311,7 +311,7 @@ object FormSpec extends ZIOHttpSpec {
           )
         }
       } @@ samples(10),
-    )
+    ) @@ sequential
```
Comment (kyri-petrou) [inline zio-http/jvm/src/test/scala/zio/http/FormSpec.scala:314]: Marked this test suite as sequential hoping it'll fix the flakiness

**Comment by kyri-petrou** [comment]
@987Nabil the test issue is gone now that I made the test suite run sequentially, but the coverage action is hanging due to an issue with the CI setup:

```
{'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
```

**Pull request merged**